apiserver/facades/controller/undertaker/undertaker_test.go

```diff
@@ -27,15 +29,23 @@ type undertakerSuite struct {
 	coretesting.BaseSuite
 	secrets                  *mockSecrets
 	mockSecretBackendService *MockSecretBackendService
-	modelConfigService       *MockModelConfigService
+	mockModelConfigService   *MockModelConfigService
+	mockModelInfoService     *MockModelInfoService
+	ctx                      context.Context
```

Comment:
minor nit: I think the ctx here could be refactored to individual methods instead of sharing them for each test. Although it might not really matter much in the current test cases, but might lead to unexpected issues if we continue to reuse the context, for eg cancelling.  


Suggestion for each test, it can look something like 

TestXXX () {
        
        ctx := context.Background()
	err = hostedAPI.ProcessDyingModel(ctx)
	....
	err = hostedAPI.ProcessDyingModel(ctx)

}

Reply:
fixed.